components/pages/blog/list.js

```diff
@@ -55,33 +61,45 @@ const Blog = () => {
             console.error('Error deleting post:', error);
         }
     };
+
     const renderAddPostButton = () => {
         if (authorized) {
             return (
-                <Button variant="outlined" color="secondary" onClick={() => openEditModalHandler({})}>
+                <Button variant="outlined" color="secondary" onClick={openAddModalHandler}>
                     Add Post
                 </Button>
             );
         }
         return null;
     };
+
     return (
         <Container>
             <Title>Welcome to My Blog</Title>
             {renderAddPostButton()}
             <PostListComponent
                 posts={posts}
                 isButtons={authorized}
-                openEditModalHandler={openEditModalHandler}
-                deletePost={handleDeletePost}
-            />
-            <PostModal
-                isModalOpen={isModalOpen}
-                closeModalHandler={closeModalHandler}
-                editedPost={editedPost}
                 handleEditPostAsync={handleEditPostAsync}
-                handleAddPost={handleAddPost}
+                deletePost={handleDeletePost}
+                openEditModalHandler={openEditModalHandler}
             />
+                <PostModal
+                    title={'Add post'}
+                    isModalOpen={isAddModalOpen}
+                    closeModalHandler={closeAddModalHandler}
+                    editedPost={{}}
```

Comment:
You can just remove `editedPost={{}}` from props and handle the `undefined` value inside of the component